### PR TITLE
Add GCP PD CSI driver addon

### PR DESCRIPTION
**Pull request by rifelpet**
https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver

Kubernetes 1.23 enables `CSIMigrationGCE` feature flag by default (see [changelog](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.23.md)) which requires these components be deployed. This will (eventually) fix the sig-storage test failures in the periodic job: https://testgrid.k8s.io/kops-gce#kops-gce-latest

### CODE REVIEW

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:261]: I do not see this namespace to be created anywhere?

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:275]: same here

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:291]: needs namespace that kops can put it to correct namespace

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:457]: missing namespace

**Review by zetaab — CHANGES_REQUESTED** [review]
overall looks quite good. However, 2 minor issues and then this PR is missing changes to storageclasses.

See https://github.com/zetaab/kops/blob/gcpcsi/upup/models/cloudup/resources/addons/storage-gce.addons.k8s.io/k8s-1.17.yaml.template#L2-L16

and then remove 

```
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
```
from existing storageclass.

I had PR open https://github.com/kubernetes/kops/pull/12736/files already but we can merge this one if you fix those few things

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:424]: this manifest does not have `latest` tag available, so it will not work like this.

https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver says that the latest image is `k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver:v1.3.4`

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:405]: should we bind these versions to some tag? latest can break things in production

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:376]: should we bind these versions to some tag? latest can break things in production

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:347]: should we bind these versions to some tag? latest can break things in production

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:313]: should we bind these versions to some tag? latest can break things in production

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:478]: should we bind these versions to some tag? latest can break things in production

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:496]: `k8s.gcr.io/cloud-provider-gcp/gcp-compute-persistent-disk-csi-driver:v1.3.4`

**Review by zetaab — CHANGES_REQUESTED** [review]

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:427]: add

```
            - "--extra-labels=k8s-io-cluster-name={{ replace ClusterName "." "-" }}"
```

File: upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template
Comment (zetaab) [inline upup/models/cloudup/resources/addons/gcp-pd-csi-driver.addons.k8s.io/k8s-1.23.yaml.template:428]: this whole csi driver does not work if you specify this env, you need to remove it (and you can remove that /etc/cloud-sa volumemount as well)

File: tests/integration/update_cluster/minimal_gce/data/aws_s3_bucket_object_minimal-gce.example.com-addons-gcp-pd-csi-driver.addons.k8s.io-k8s-1.23_content
Comment (zetaab) [inline tests/integration/update_cluster/minimal_gce/data/aws_s3_bucket_object_minimal-gce.example.com-addons-gcp-pd-csi-driver.addons.k8s.io-k8s-1.23_content:1]: should we use `{{ if WithDefaultBool .CloudConfig.ManageStorageClasses true }}` here? If someone want disable the default behaviour.

File: tests/integration/update_cluster/minimal_gce/data/aws_s3_bucket_object_minimal-gce.example.com-addons-gcp-pd-csi-driver.addons.k8s.io-k8s-1.23_content
Comment (zetaab) [inline tests/integration/update_cluster/minimal_gce/data/aws_s3_bucket_object_minimal-gce.example.com-addons-gcp-pd-csi-driver.addons.k8s.io-k8s-1.23_content:697]: use 

```
            - "--extra-labels=k8s-io-cluster-name={{ replace ClusterName "." "-" }}"
```

**Review by zetaab — CHANGES_REQUESTED** [review]

**Review by zetaab — APPROVED** [review]
looks good!

**Comment by zetaab** [comment]
/hold

do you want @rifelpet fix those first or merge this?

**Comment by zetaab** [comment]
still some tests failing

**Comment by rifelpet** [comment]
Yes the two relevant ones are regarding topologies, it seems the tests are using the wrong topology label, I've been meaning to look into how to configure the desired label in the test suite, using the PD CSI repo's own e2e tests as an example

**Comment by rifelpet** [comment]
I've decided to skip the failing topology tests. There doesn't seem to be a way to use the CSI driver's topology keys in the e2e in-tree tests. This brings up a larger question of whether we should be running the in-tree storage tests at all when using an external CSI driver.

I'm giving this one last go. Once we confirm only the (unrelated) metadata concealment test is failing, I'll force push once more to clear the GH status context and then we can merge.

/test pull-kops-e2e-k8s-gce-ci

**Pull request merged**
